pkg/model/utxo/treasury_output.go

```diff
@@ -0,0 +1,246 @@
+	// Do not modify the value since we're writing this as a bool.
+	TreasuryOutputUnspentPrefix = 0
+)
+
+var (
+	// Returned when the state of the treasury is invalid.
+	ErrInvalidTreasuryState = errors.New("invalid treasury state")
+)
+
+// TreasuryOutput represents the output of a treasury transaction.
+type TreasuryOutput struct {
+	// The ID of the milestone which generated this output.
+	MilestoneID iotago.MilestoneID
+	// The amount residing on this output.
+	Amount uint64
+	// Whether this output was already spent
+	Spent bool
+}
+
+func (t *TreasuryOutput) kvStorableKey() (key []byte) {
+	return marshalutil.New(34).
+		WriteByte(UTXOStoreKeyPrefixTreasuryOutput).
+		WriteBool(t.Spent).
+		WriteBytes(t.MilestoneID[:]).
+		Bytes()
+}
+
+func (t *TreasuryOutput) kvStorableValue() (value []byte) {
+	return marshalutil.New(8).
+		WriteUint64(t.Amount).
+		Bytes()
+}
+
+func (t *TreasuryOutput) kvStorableLoad(_ *Manager, key []byte, value []byte) error {
+	keyExt := marshalutil.New(key)
+	// skip prefix
+	if _, err := keyExt.ReadByte(); err != nil {
+		return err
+	}
+
+	spent, err := keyExt.ReadBool()
+	if err != nil {
+		return err
+	}
+
+	milestoneID, err := keyExt.ReadBytes(iotago.MilestoneIDLength)
+	if err != nil {
+		return err
+	}
+	copy(t.MilestoneID[:], milestoneID)
+
+	val := marshalutil.New(value)
+	t.Amount, err = val.ReadUint64()
+	if err != nil {
+		return err
+	}
+
+	t.Spent = spent
+
+	return nil
+}
+
+// stores the given treasury output.
+func storeTreasuryOutput(output *TreasuryOutput, mutations kvstore.BatchedMutations) error {
+	return mutations.Set(output.kvStorableKey(), output.kvStorableValue())
+}
+
+// deletes the given treasury output.
+func deleteTreasuryOutput(output *TreasuryOutput, mutations kvstore.BatchedMutations) error {
+	return mutations.Delete(output.kvStorableKey())
+}
+
+// marks the given treasury output as spent.
+func markTreasuryOutputAsSpent(output *TreasuryOutput, mutations kvstore.BatchedMutations) error {
+	outputCopy := *output
+	outputCopy.Spent = false
+	if err := mutations.Delete(outputCopy.kvStorableKey()); err != nil {
+		return err
+	}
+	outputCopy.Spent = true
+	return mutations.Set(outputCopy.kvStorableKey(), outputCopy.kvStorableValue())
+}
+
+// marks the given treasury output as unspent.
+func markTreasuryOutputAsUnspent(output *TreasuryOutput, mutations kvstore.BatchedMutations) error {
+	outputCopy := *output
+	outputCopy.Spent = true
+	if err := mutations.Delete(outputCopy.kvStorableKey()); err != nil {
+		return err
+	}
+	outputCopy.Spent = false
+	return mutations.Set(outputCopy.kvStorableKey(), outputCopy.kvStorableValue())
+}
+
+func (u *Manager) readSpentTreasuryOutputWithoutLocking(msHash []byte) (*TreasuryOutput, error) {
+	key := append([]byte{UTXOStoreKeyPrefixTreasuryOutput, TreasuryOutputSpentPrefix}, msHash...)
+	val, err := u.utxoStorage.Get(key)
+	if err != nil {
+		return nil, err
+	}
+	to := &TreasuryOutput{}
+	if err := to.kvStorableLoad(nil, key, val); err != nil {
```

Comment:
I would still pass `u` here. If for some reason we use it in the future we will not understand why it is not set.